src/main/java/org/example/parampammusic/service/UserService.java

```diff
@@ -0,0 +1,61 @@
+package org.example.parampammusic.service;
+
+import jakarta.servlet.http.HttpSession;
+import lombok.RequiredArgsConstructor;
+import org.example.parampammusic.entity.Role;
+import org.example.parampammusic.entity.User;
+import org.example.parampammusic.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Service
+@RequiredArgsConstructor
+public class UserService implements UserDetailsService {
+
+    @Autowired
```

Comment:
Use the same approach for all classes.
In other places you use constructors, but here - @Autowired annotation.
Choose the same approach!